lib/Plack/Middleware/AccessLog/Structured.pm

```diff
@@ -65,9 +63,9 @@ sub call {
 			content_length   => Plack::Util::content_length($cb_res->[2]) || $h->get('Content-Length'),
 			content_type     => defined $content_type ? "$content_type" : undef,
 			# Timing
-			request_duration => ( $t_after - $t_before ) * 1000,
-			date             => DateTime->from_epoch(epoch => $t_before)->strftime('%Y-%m-%dT%H:%M:%S.%3NZ'),
-			epochtime        => $t_before,
+			request_duration => ( $t_before->delta_microseconds($t_after) / 1000 ),
+			date             => $t_before->strftime('%FT%T%3fZ'),
+			epochtime        => $t_before->epoch,
```

Comment:
Compared to the original implementation, `epochtime` has only seconds precision now, `Time::HiRes` returned floating seconds since the epoch, so adding something like `$t_before->millisecond / 1000` to the `$t_before->epoch` value should improve precision again and actually represent the same timestamp as `date`.

Reply:
Done in update